### PR TITLE
Eventhub : implemented exception handler

### PR DESCRIPTION
**Pull request by v-Ajnava**
---
* exception handler implemented
* fixed issue found in eventhub update command

This checklist is used to make sure that common guidelines for a pull request are followed.

- [X] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [X] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/command_modules/azure-cli-eventhubs/azure/cli/command_modules/eventhubs/_exception_handler.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+            raise CLIError(ex)
+    else:
+        import sys
+        from six import reraise
+        reraise(*sys.exc_info())
+
+
+# pylint: disable=inconsistent-return-statements
+def empty_on_404(ex):
+    from azure.mgmt.eventhub.models import ErrorResponseException
```
Comment (tjprescott) [inline src/command_modules/azure-cli-eventhubs/azure/cli/command_modules/eventhubs/_exception_handler.py]: @williexu can confirm, but I believe we no longer require/desire using `empty_on_404`.

Comment (williexu) [reply src/command_modules/azure-cli-eventhubs/azure/cli/command_modules/eventhubs/_exception_handler.py]: We don't and we should not without good reason.

File: src/command_modules/azure-cli-eventhubs/azure/cli/command_modules/eventhubs/commands.py [hunk 2]
```diff
@@ -6,47 +6,44 @@
...
+        client_factory=disaster_recovery_mgmt_client_factory)
 
 # Namespace Region
     custom_tmpl = 'azure.cli.command_modules.eventhubs.custom#{}'
     eventhubs_custom = CliCommandType(operations_tmpl=custom_tmpl)
-    with self.command_group('eventhubs namespace', eh_namespace_util, client_factory=namespaces_mgmt_client_factory) as g:
+    with self.command_group('eventhubs namespace', eh_namespace_util, client_factory=namespaces_mgmt_client_factory, exception_handler=response_exception_handler) as g:
         g.custom_command('create', 'cli_namespace_create')
         g.show_command('show', 'get')
         g.custom_command('list', 'cli_namespace_list', exception_handler=empty_on_404)
```
Comment (williexu) [inline src/command_modules/azure-cli-eventhubs/azure/cli/command_modules/eventhubs/commands.py]: @v-Ajnava please remove the `empty_on_404` behavior for list commands. They should fail like our other list commands.

File: src/command_modules/azure-cli-eventhubs/azure/cli/command_modules/eventhubs/custom.py [hunk 3]
```diff
@@ -104,28 +104,25 @@ def cli_eheventhub_update(instance, message_retention_in_days=None, partition_co
...
 
-    if instance.enabled is True or enabled is True:
+    if enabled is True:
+        if instance.capture_description is None:
+            instance.capture_description = CaptureDescription()
+            instance.capture_description.destination = Destination()
+            instance.capture_description.encoding = EncodingCaptureDescription.avro
+
         if enabled:
             instance.capture_description.enabled = enabled
```
Comment (williexu) [inline src/command_modules/azure-cli-eventhubs/azure/cli/command_modules/eventhubs/custom.py]: You already check that `enabled` is `True` in the outer nesting. You can remove the `if` and move this up.

File: src/command_modules/azure-cli-eventhubs/azure/cli/command_modules/eventhubs/commands.py (hunk 2 above)
Comment (v-Ajnava) [reply src/command_modules/azure-cli-eventhubs/azure/cli/command_modules/eventhubs/commands.py]: resolved

File: src/command_modules/azure-cli-eventhubs/azure/cli/command_modules/eventhubs/custom.py (hunk 3 above)
Comment (v-Ajnava) [reply src/command_modules/azure-cli-eventhubs/azure/cli/command_modules/eventhubs/custom.py]: resolved

File: src/command_modules/azure-cli-eventhubs/azure/cli/command_modules/eventhubs/_exception_handler.py (hunk 1 above)
Comment (v-Ajnava) [reply src/command_modules/azure-cli-eventhubs/azure/cli/command_modules/eventhubs/_exception_handler.py]: resolved

**Comment by v-Ajnava** [comment]
@tjprescott and @williexu can you please review the changes

File: src/command_modules/azure-cli-eventhubs/HISTORY.rst [hunk 4]
```diff
@@ -3,6 +3,11 @@
 Release History
 ===============
 
+0.2.5
++++++
+* exception handling fixed
```
Comment (troydai) [inline src/command_modules/azure-cli-eventhubs/HISTORY.rst]: 'Fix exception handling'

File: src/command_modules/azure-cli-eventhubs/HISTORY.rst [hunk 5]
```diff
@@ -3,6 +3,11 @@
...
+* fixed eventhub update command
```
Comment (troydai) [inline src/command_modules/azure-cli-eventhubs/HISTORY.rst]: 'Fix eventhub update command'

File: src/command_modules/azure-cli-eventhubs/azure/cli/command_modules/eventhubs/custom.py [hunk 6]
```diff
@@ -104,28 +104,25 @@ def cli_eheventhub_update(instance, message_retention_in_days=None, partition_co
     if status:
         instance.status = status
 
-    if instance.enabled is True or enabled is True:
-        if enabled:
-            instance.capture_description.enabled = enabled
+    if enabled is True:
```
Comment (troydai) [inline src/command_modules/azure-cli-eventhubs/azure/cli/command_modules/eventhubs/custom.py]: `if enable`

File: src/command_modules/azure-cli-eventhubs/azure/cli/command_modules/eventhubs/custom.py [hunk 7]
```diff
@@ -104,28 +104,25 @@ def cli_eheventhub_update(instance, message_retention_in_days=None, partition_co
...
+        instance.capture_description.enabled = enabled
+
+        if instance.capture_description is None:
```
Comment (troydai) [inline src/command_modules/azure-cli-eventhubs/azure/cli/command_modules/eventhubs/custom.py]: `if not instance.capture_description:`

File: src/command_modules/azure-cli-eventhubs/azure/cli/command_modules/eventhubs/custom.py (hunk 6 above)
Comment (v-Ajnava) [reply src/command_modules/azure-cli-eventhubs/azure/cli/command_modules/eventhubs/custom.py]: resolved

File: src/command_modules/azure-cli-eventhubs/HISTORY.rst (hunk 5 above)
Comment (v-Ajnava) [reply src/command_modules/azure-cli-eventhubs/HISTORY.rst]: Resolved

File: src/command_modules/azure-cli-eventhubs/HISTORY.rst (hunk 4 above)
Comment (v-Ajnava) [reply src/command_modules/azure-cli-eventhubs/HISTORY.rst]: Removed, has reverted the exception handling changes as core now handles bad request

**Comment by troydai** [comment]
Fixes CI plese.

**Comment by v-Ajnava** [comment]
@troydai , updated the histor, minor version bump for breaking change due to removing exception handler for 404 list commands. as suggested in ServiceBus PR: https://github.com/Azure/azure-cli/pull/7265

**Comment by troydai** [comment]
Looks good to me. I will wait for @williexu

**Pull request merged**